RadarSDK/Radar.m

```diff
@@ -198,20 +198,19 @@ + (void)trackOnceWithDesiredAccuracy:(RadarTrackingOptionsDesiredAccuracy)desire
                                            beacons:beacons
                                  completionHandler:^(RadarStatus status, NSDictionary *_Nullable res, NSArray<RadarEvent *> *_Nullable events, RadarUser *_Nullable user,
                                                      NSArray<RadarGeofence *> *_Nullable nearbyGeofences, RadarConfig *_Nullable config, RadarVerifiedLocationToken *_Nullable token) {
-                                     if (status == RadarStatusSuccess) {
-                                         [[RadarLocationManager sharedInstance] replaceSyncedGeofences:nearbyGeofences];
-                                         if (config != nil) {
-                                             [[RadarLocationManager sharedInstance] updateTrackingFromMeta:config.meta];
-                                         }
-                                         
-                                     }
-
-                                     if (completionHandler) {
-                                         [RadarUtils runOnMainThread:^{
-                                             completionHandler(status, location, events, user);
-                                         }];
-                                     }
-                                 }];
+                                    if (config != nil) {
```

Comment:
This is one of the changes I highlighted regarding the track lifecycle. When the server track request fails, we fallback to the offline manager to act as a stand-in. In these cases, although we return a non-success status, we still receive a valid config, which the rest of the track lifecycle should process.

Therefore, instead of relying on the status code, we should check if the config is non-nil. A non-nil config only occurs either from a successful server-side track or when the offline manager handles the network error.